### PR TITLE
[pkg/stanza/operator/transformer/recombine] docs: add example configurations with max_unmatched_batch_size

### PR DESCRIPTION
**Pull request by kasia-kujawa**
**Description:** add example configurations for recombine operator with `max_unmatched_batch_size`

**Link to tracking Issue:** #31653

### CODE REVIEW

**Comment by kkujawa-sumo** [comment]
@sumo-drosiek @astencel-sumo please take a look

File: pkg/stanza/docs/operators/recombine.md [hunk 1]
```diff
@@ -141,3 +141,146 @@ The following logs will be output:
   },
 ]
 ```
+
+#### Example configurations with `max_unmatched_batch_size`
+
+##### `max_unmatched_batch_size=0`
```
Comment (sumo-drosiek) [inline pkg/stanza/docs/operators/recombine.md:147]:
```suggestion
##### `max_unmatched_batch_size` set to `0`
```

It sounds better for me when I read it 😅 

And the same for all cases

Comment (kkujawa-sumo) [reply pkg/stanza/docs/operators/recombine.md]: I changed it :)

**Review by sumo-drosiek — APPROVED** [review]
Thanks!

**Review by andrzej-stencel — APPROVED** [review]

**Pull request merged**